vyper/venom/analysis/mem_ssa.py

```diff
@@ -0,0 +1,439 @@
+                    return None  # Found a use that reads from our memory location
+            if clobber is not None:
+                return clobber
+
+        # Traverse successors
+        worklist = list(block.cfg_out)
+        visited = {block}
+        while worklist:
+            succ = worklist.pop()
+            if succ in visited:
+                continue
+            visited.add(succ)
+
+            # Check phi nodes
+            if succ in self.memory_phis:
+                phi = self.memory_phis[succ]
+                for op_def, pred in phi.operands:
+                    if pred == block and op_def == access:
+                        # This def reaches the phi, check if phi is clobbered
+                        for inst in succ.instructions:
+                            next_def = self.inst_to_def.get(inst)
+                            if next_def and next_def.loc.completely_contains(def_loc):
+                                return next_def
+                            mem_use = self.inst_to_use.get(inst)
+                            if mem_use and mem_use.loc.completely_contains(def_loc):
+                                return None  # Found a use that reads from our memory location
+
+            # Check instructions in successor block
+            for inst in succ.instructions:
+                next_def = self.inst_to_def.get(inst)
+                if next_def and next_def.loc.completely_contains(def_loc):
+                    return next_def
+                mem_use = self.inst_to_use.get(inst)
+                if mem_use and mem_use.loc.completely_contains(def_loc):
+                    return None  # Found a use that reads from our memory location
+
+            worklist.extend(succ.cfg_out)
+
+        return None
+
+    #
+    # Printing context methods
+    #
+    def _post_instruction(self, inst: IRInstruction) -> str:
+        s = ""
+        if inst.parent in self.memory_uses:
+            for use in self.memory_uses[inst.parent]:
+                if use.load_inst == inst:
+                    s += f"\t; use: {use.reaching_def.id_str if use.reaching_def else None}"
+        if inst.parent in self.memory_defs:
+            for def_ in self.memory_defs[inst.parent]:
+                if def_.store_inst == inst:
+                    s += f"\t; def: {def_.id_str} "
+                    s += f"({def_.reaching_def.id_str if def_.reaching_def else None}) "
+                    s += f"{self.get_clobbering_memory_access(def_)}"
+
+        return s
+
+    def _pre_block(self, bb: IRBasicBlock) -> str:
+        s = ""
+        if bb in self.memory_phis:
+            phi = self.memory_phis[bb]
+            s += f"    ; phi: {phi.id_str} <- "
+            s += ", ".join(f"{op[0].id_str} from @{op[1].label}" for op in phi.operands)
+            s += "\n"
+        return s
+
+    @contextlib.contextmanager
+    def print_context(self):
+        ir_printer.set(self)
+        yield
```

Comment:
probably doesn't matter too much but yield is not exception safe so in general it's best practice to protect it with try/finally --
```suggestion
        try:
            yield
        finally:
```